### PR TITLE
[Spotless] Adds new line at end of java files

### PR DESCRIPTION
**Pull request by MitchellGale**
### Description
Adds new line at end of files missing it. Uncommented spotless line for verification.

**Note:** line endings updated from `/r/n` to `/n` in `common/src/main/java/org/opensearch/sql/common/grok/Converter.java`
  
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by GumpacG — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -83,18 +83,18 @@ repositories {
 // Spotless checks will be added as PRs are applied to resolve each style issue is approved.
 spotless {
     java {
-//        target fileTree('.') {
-//            include '**/*.java', 'src/*/java/**/*.java'
-//            exclude '**/build/**', '**/build-*/**'
-//        }
+        target fileTree('.') {
+            include '**/*.java', 'src/*/java/**/*.java'
```
Comment (Yury-Fridlyand) [inline build.gradle:87]: We don't have `src` in the repo root, I think one condition is enough
```suggestion
            include '**/*.java'
```

Comment (Yury-Fridlyand) [reply build.gradle:87]: But we can cover `json` files there

**Review by Yury-Fridlyand — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (MitchellGale) [reply build.gradle]: Thanks, removed

File: core/src/main/java/org/opensearch/sql/ast/tree/RareTopN.java [hunk 2]
```diff
@@ -57,4 +57,3 @@ public enum CommandType {
     RARE
   }
 }
-
```
Comment (matthewryanwells) [inline core/src/main/java/org/opensearch/sql/ast/tree/RareTopN.java:60]: Stupid question but this PR mentions adding newlines at the end of files but I've noticed that it looks like this PR removes newlines, am I misunderstanding what you mean by newlines?

Comment (MitchellGale) [reply core/src/main/java/org/opensearch/sql/ast/tree/RareTopN.java:60]: image
It appears weird on GitHub, but if you look at the change in an editor it is correct. In the case of RareTopN.java it is removing a new line as it ends with two new lines.

So, yes, it does remove it if more than 1 new line exists.

**Comment by Yury-Fridlyand** [comment]
Do you want to extend this feature to json files too? (we have number of files in tests).
And for docs.

File: core/src/main/java/org/opensearch/sql/ast/tree/RareTopN.java (hunk 2 above)
Comment (matthewryanwells) [reply core/src/main/java/org/opensearch/sql/ast/tree/RareTopN.java:60]: Ah okay thanks for letting me know, just Github being weird

**Review by matthewryanwells — APPROVED** [review]

**Comment by MitchellGale** [comment]
> Do you want to extend this feature to json files too? (we have number of files in tests). And for docs.
@Yury-Fridlyand 
I think we should add some of these to json and other files in a future PR. For now I think we should get the changes over for just java files and can address other files later.

**Pull request merged**